README.mdx

```diff
@@ -0,0 +1,198 @@
+
+Think of composable architectures like [building with legos](https://twitter.com/cdixon/status/1450462593232089088?s=20) 
+- you build something you need as a block or set of blocks and someone else can come along and use those same blocks to 
+build something entirely new.
+
+Composability on Flow, especially with attachments is permissionless because not only can you build with existing 
+components, but no one can stop you from doing so!
+
+</Callout>
+
+Permissionless composability is huge because, as builders, we can build on existing constructs for existing 
+communities and be sure that the original developer won't rug our access - AKA platform risk.
+
+For an example of the negative case, just look at how big tech companies locked down their API in the early days.
+
+Less duplicate code, easier to bootstrap attention and adoption, and users benefit from the utility you built into 
+something they already presumably own and love - win-win-win!
+
+# Define our Base NFT - KittyVerse
+
+## NonFungibleToken Standard
+
+- Building on the existing standard
+- Encouraged to also implement MetadataViews and ViewResolver contract interfaces, but keeping things simple
+- We're defining the KittyVerse NFT for the purposes of this tutorial, but just as easily could've used an existing NFT 
+project since we're building with attachments
+- If totally new to this, take a look at NonFungibleToken tutorials - TODO: Link these
+
+## Contract Overview
+
+- Contract interface implementation
+    - Events
+    - Nested resource definitions
+- Attribute overview
+- Note on `unsafeRandom()`
+- Opportunities to extend the contract
+
+## Deploy KittyVerse contract
+
+- Deploy NonFungibleToken to `0x01`
+- Deploy KittyVerse to `0x02`
+
+## Setup account & Mint KittyVerse NFT
+
+- send `tx_01_setup_kittyverse_account.cdc` signing with `0x02`
+- send `tx_02_mint_kittyverse_nft.cdc` signing with `0x02`
+- send `tx_02_mint_kittyverse_nft.cdc` again signing with `0x02`
+- query NFT.id - execute `script_01_get_kittyverse_ids.cdc` providing `0x02` as argument
+    - see two unique NFT IDS - take note of them, next let's prep them some hats
+
+# Define our Attachments - KittyHats
+
+## Attachments
+
+- What are they?
+    - In Cadence
+    - Simple mental model
+        - "Like glue between resources"
+        - ![KittyVerse & KittyHats attached](./diagrams/cat_with_hat.png)
+- Why do they matter?
+    - Can create attachments for anything without needing original developer's:
+        - Participation
+        - Permission
+
+### Overview
+
+- Conceptual
+    - Attachments are a special type that can be added to resources (doc link needed)
+    - You might be tempted to simply define all your functionality in the attachment, but you'll lose the existence and
+      uniqueness guarantees inherent to resources as well as make access control significantly harder for yourself
+        - As mentioned earlier, attachments are best thought of as glue between resources
```

Comment:
Maybe we should be explicit and also say that they aren't Resources and shouldn't be thought of that way? Maybe also we should explain that it's utility is not for use when within ones own contracts, but when other contracts want to interact and contribute something new to an existing Resource

Reply:
> Maybe we should be explicit and also say that they aren't Resources and shouldn't be thought of that way?
Agreed, worth noting that fact
> Maybe also we should explain that it's utility is not for use when within ones own contracts, but when other contracts want to interact and contribute something new to an existing Resource
IDK if we need to be that prescriptive about their use. It's not clear to me that it's necessarily an anti-pattern to develop with attachments within contracts you own. Is there a reason you think it's bad practice?

Reply:
It would be better not to have the attachment at the very first place. until I know this attachment would be useful to some x-audience. So it would be easy to manage and less error-prone. While we can add attachments at any time during upgrades, isn't it?

Reply:
> It's not clear to me that it's necessarily an anti-pattern to develop with attachments within contracts you own. Is there a reason you think it's bad practice

Not entirely sure myself, it's something that @joshuahannan spoke to more. He argued before that if it's your contract you don't need attachments since you can explicitly design the functionality into the main contract. Maybe we should make sure the documentation makes it clear that providing support for attachments in your Resources is a key way to enable composability from unknown future sources because it signals that the project is open to collaboration/contributions